src/scripts/components/widgets/modals/ModalStake.js

```diff
@@ -91,60 +101,75 @@ class ModalStake extends Component<Props, Object> {
       })
   }
 
+  componentWillReceiveProps (nextProps: Props): void {
+    const selectedVideo = nextProps.selectedVideo
+    this.setState({
+      id: selectedVideo.id,
+      title: selectedVideo.title,
+      description: selectedVideo.description,
+      duration: selectedVideo.duration,
+      author: selectedVideo.author,
+      errorMessage: false,
+      agreedTOC: false // TODO,
+    })
```

Comment:
Just want to understand better. Why is all of this put in state when it can just be read from props?

Reply:
@ya7ya could we change that?